connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,99 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Test;
+
+import java.io.*;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
+    private static Connection connection;
```

Comment:
&#x1F34E; **Issue**
If you want to have a state in your class this state should not be static (it should be local to the class in question - that is it should be a member field). With statics, well, you share it between tests and this might result in some test leaking its data into another.
Statics are usually only used for:
1) Static procedure (it doesn't depend on any state, just does something based on its arguments).
2) Static final constant.
3) Singletones.